CMakeLists.txt

```diff
@@ -24,35 +24,373 @@
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
-endif(CCACHE_FOUND)
+INCLUDE ( CMakeDependentOption )
 
+# @todo SSE/AVX instruction options
+# @todo static/shared library options
+OPTION ( OPENVDB_BUILD_CORE "Build the OpenVDB core" ON )
+OPTION ( OPENVDB_BUILD_BINARIES "Build the vdb binaries" ON )
+OPTION ( OPENVDB_BUILD_PYTHON_MODULE "Build the pyopenvdb Python module" ON )
 OPTION ( OPENVDB_BUILD_UNITTESTS "Build the OpenVDB unit tests" ON )
 OPTION ( OPENVDB_BUILD_DOCS "Build the OpenVDB documentation" OFF )
-OPTION ( OPENVDB_BUILD_PYTHON_MODULE "Build the pyopenvdb Python module" ON )
 OPTION ( OPENVDB_BUILD_HOUDINI_SOPS "Build the Houdini SOPs" OFF )
 OPTION ( OPENVDB_BUILD_MAYA_PLUGIN "Build the Maya plugin" OFF )
-OPTION ( OPENVDB_BUILD_CORE "Build the OpenVDB core" ON )
 OPTION ( OPENVDB_ENABLE_RPATH "Build with RPATH information" ON )
-OPTION ( OPENVDB_ENABLE_3_ABI_COMPATIBLE "Build with OpenVDB v3 ABI" ON )
-OPTION ( OPENVDB_DISABLE_BOOST_IMPLICIT_LINKING "Disable the implicit linking of Boost libraries on Windows" ON )
 OPTION ( OPENVDB_CXX_STRICT "Enable or disable a set of pre-defined compiler warnings for clang and gcc" OFF )
+OPTION ( OPENVDB_ENABLE_3_ABI_COMPATIBLE "Build with OpenVDB ABI 3 (deprecated - see OPENVDB_ABI_VERSION_NUMBER)" OFF )
+
+###### Dependency options
+
+# @todo scalable/concurrent malloc option(s)
+# @todo non-required behavior for blosc, log4cplus (i.e. use if available)
+# @todo OPENVDB_USE_NUMPY options for NumPy
+OPTION ( USE_HOUDINI [=[
+Build the library against a Houdini installation. Turns on autmatically if OPENVDB_BUILD_HOUDINI_SOPS is enabled.
+When enabled, you do not need to provide dependency locations for OpenEXR, IlmBase, TBB, Boost and Blosc. You can
+optionally provide your own version of Boost when building against Houdini 16.5 and gerater.]=] OFF )
+OPTION ( USE_BLOSC "Build the library with blosc support." ON )
+OPTION ( USE_LOG4CPLUS "Build the library with log4cplus support." OFF )
+OPTION ( OPENEXR_NAMESPACE_VERSIONING "Set the expected names of OpenEXR libraries to be version suffixed" OFF )
+OPTION ( ILMBASE_NAMESPACE_VERSIONING "Set the expected names of ILM Base libraries to be version suffixed" OFF )
+CMAKE_DEPENDENT_OPTION ( OPENVDB_DISABLE_BOOST_IMPLICIT_LINKING
+  "Disable the implicit linking of Boost libraries on Windows" ON "WIN32" OFF)
 OPTION ( USE_SYSTEM_LIBRARY_PATHS "Build with system library paths" ON )
+CMAKE_DEPENDENT_OPTION ( BOOST_PYTHON_NAMESPACE_VERSIONING
+  "Set the expected names of Boost.Python libraries to be version suffixed"
+  ON "OPENVDB_BUILD_PYTHON_MODULE" ON )
+
+SET ( OPENVDB_ABI_VERSION_NUMBER "" CACHE STRING "Build for compatibility with version N of the
+  OpenVDB Grid ABI, where N is 3, 4, 5 etc. (some newer features will be disabled)." )
+
+# Various dependency library versions - Cache options can be changed
+
+SET ( MINIMUM_BOOST_VERSION 1.55 )
+SET ( MINIMUM_BLOSC_VERSION 1.5 )
+SET ( MINIMUM_GLFW_VERSION 3 )
+
```

Comment:
It might be nice to add something like "`SET ( MINIMUM_OPENVDB_ABI_VERSION 3 )`" here (and some appropriate range checking logic down in the ABI section), if we're going to start dropping support for older ABIs soon.

Reply:
Done with a resulting warning